client/ayon_core/hosts/houdini/plugins/create/create_model.py

```diff
@@ -0,0 +1,144 @@
+# -*- coding: utf-8 -*-
+"""Creator plugin for creating Model product type.
+
+Note:
+    Currently, This creator plugin is the same as 'create_pointcache.py'
+    But renaming the product type to 'model'.
+
+    It's purpose to support
+    Maya (load/publish model from maya to/from houdini).
+
+    It's considered to support multiple representations in the future.
+"""
+
+from ayon_core.hosts.houdini.api import plugin
+from ayon_core.lib import BoolDef
+
+import hou
+
+
+
+class CreateModel(plugin.HoudiniCreator):
+    """Create Model"""
+    identifier = "io.openpype.creators.houdini.model"
+    label = "Model"
+    product_type = "model"
+    icon = "cube"
+
+    def get_publish_families(self):
+        return ["model", "abc"]
```

Comment:
I wonder whether this is the best solution here. @iLLiCiTiT thoughts?
Or should we do the reverse by e.g. adding the family `model` where we want it to apply instead, similar to e.g. [this](https://github.com/ynput/ayon-core/pull/212/files#diff-6543b87b81db3323d73f825c3cd13d657922b73f812c71b777dcd91ee869630eR82-R90)?

Reply:
> Or should we do the reverse by e.g. adding the family model where we want it to apply instead, similar to e.g. [this](https://github.com/ynput/ayon-core/pull/212/files#diff-6543b87b81db3323d73f825c3cd13d657922b73f812c71b777dcd91ee869630eR82-R90)?

I personally prefer this, and I applied similar ways in Max host to avoid unnecessary duplicates of the codes. For this case, we can consider doing that if it is just for publishing pointcache model in abc format, as I can see that most of the codes are similar to the one in the creator. Guess we can just refactor the pointcache creator code a little bit, (e.g. put those parameters which sets abc to be static or dynamic into separate function). and the model creator inherited from it. 
But anyways, I am not sure about the scope for Houdini and there might be a reason why applying `get_published families` instead of having child class to inherit from the similar extractor

Reply:
Let me add my 2 cents, please correct me if I'm wrong. 

I think by adding the following snippet to `extract_alembic.py` , it should replace `extract_model.py` (it doesn't exist.)

The extractor will work but it'd be redundant because having `abc` in families is sufficient.
Also, collectors and validators associated with alembic won't be triggered. 

```python
class ExtractModelABC(ExtractAlembic):
    """Extract model as ABC."""

    label = "Extract Alembic (Model)"
    families = ["model"]

    # Driven by settings
    optional = True
```

But, the current solution to add `abc` to families is by using such a collector. 
tbh, I find it redundant if we are capable of specifying publish families inside the creators. 
e.g. in alembic creators, it would be `["pointcache", "abc"]` and in bgeo `["pointcache", "bgeo"]`

https://github.com/ynput/ayon-core/blob/e1f1556ce2c6ab5a15fda488f6af7605ca875867/client/ayon_core/hosts/houdini/plugins/publish/collect_pointcache_type.py#L9-L21


and for model creator. `["model", "abc"]` should be sufficient.

Reply:
Correct - but there is a current major disadvantage to that approach.

The **new publisher** will only show instance attribute definitions (the editable attributes) for the `product_type` set by the Creator, not by the publish families. So it will show ONLY those for the `model` family, not for the `Extractor` that is targeting only the `abc`.

In this sense the instance attribute definitions (currently) also aren't 'dynamic' per instance, but solely based on the Creator's `product_type` attribute. So if say, the creator would have an option to enable/disable the `abc` family (to enable/disable the alembic export) there would be no way to on toggle or alike update whether the Alembic export options would show or now show on the instance either (during the Create Context stage before publishing).

---

By separating it out into a separate plug-in for the extractor you also get the potential benefit that in settings you can easily specify different defaults in settings for the instance attribute definitions for both the Alembic Extractor (for pointcache) and the Alembic Extractor (for `model` family) separately.

Whether there are better "designs" that are currently possible in AYON or whether there are better ways to design the system so that it's both easy to maintain and easily customizable I'm not sure. I'd have to leave that to @dee-ynput and @iLLiCiTiT most likely.